src/python/pycold/imagetool/prepare_ard.py

```diff
@@ -1865,17 +1867,17 @@ def main(
         )
 
         if not os.path.exists(out_dir):
-            os.mkdir(out_dir)
+            os.makedirs(out_dir)
 
         if is_partition is True:
             for i in range(dataset_info.n_block_y):
                 for j in range(dataset_info.n_block_x):
                     block_folder = "block_x{}_y{}".format(j + 1, i + 1)
                     if not os.path.exists(join(out_dir, block_folder)):
-                        os.mkdir(join(out_dir, block_folder))
+                        os.makedirs(join(out_dir, block_folder))
 
         if not os.path.exists(tmp_path):
-            os.mkdir(tmp_path)
+            os.makedirs(tmp_path)
```

Comment:
I would recommend replacing this and other "if exists + makedirs" combo with the one-liner: `os.makedirs(tmp_path, exist_ok=True)`